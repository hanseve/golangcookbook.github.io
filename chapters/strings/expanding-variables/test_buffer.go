package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	var reader = bufio.NewReader(os.Stdin)
	fmt.Println("Please input a string:")
	var input, err = reader.ReadString('\n');
	if err != nil {
		fmt.Println(err)
	}
	fmt.Printf("Your input: %s\n", input)

	fmt.Println("Please input another string:")
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan()  {
		fmt.Printf("Your input: %s\n", scanner.Text())
	}
}
