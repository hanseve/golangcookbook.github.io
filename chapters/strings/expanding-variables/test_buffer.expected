Please input a string:
hello world
Your input: hello world

Please input another string:
foo bar
Your input: foo bar
hello world
Your input: hello world
